### PR TITLE
libsolv: enable zstd support to match createrepo_c

### PR DESCRIPTION
**Pull request by reubeno**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

In 3.0, `createrepo_c`'s default behavior is to create repodata files compressed using zstd; prior to this change, `libsolv` didn't support `zstd`, though, so `tdnf`/`dnf`/et al. couldn't actually handle those repos. As a workaround, the toolkit and manual invocations of `createrepo_c` for 3.0 have needed to use the `--compatibility` flag. Standard OSS tooling doesn't know to do this (nor should it have to), though; the right thing to do is to add support for zstd to `libsolv`, as other distros have done.

###### Change Log  

Enable zstd support in `libsolv` at compile time.

###### Does this affect the toolchain?  

Yes.

###### Test Methodology

Local build + pipeline build through and including toolchain (phase 1).

### CODE REVIEW

File: SPECS/libsolv/libsolv.spec [hunk 1]
```diff
@@ -10,6 +10,7 @@ Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 BuildRequires:  cmake
 BuildRequires:  rpm-devel
+BuildRequires:  libzstd-devel
```
Comment (cwize1) [inline SPECS/libsolv/libsolv.spec:13]: I suspect you might need to update the [build_official_toolchain_rpms.sh](https://github.com/microsoft/azurelinux/blob/3.0/toolkit/scripts/toolchain/build_official_toolchain_rpms.sh) file to handle the new dependency between `libsolv` and `libzstd-devel`.

Comment (cwize1) [inline SPECS/libsolv/libsolv.spec:13]: Does this increase the size of the base image? If so, by how much?

Comment (reubeno) [reply SPECS/libsolv/libsolv.spec:13]: Good question! I still need to build the artifacts to confirm, but `rpm` already has a runtime `Requires` on `libzstd.so.1()(64bit)` and that's exactly what `libsolv` ends up requiring too with this change. It seems safe to say that anywhere `libsolv` is meaningfully used in our system that `rpm` would be expected to be there too, so I would not expect to see an image size increase.

Comment (reubeno) [reply SPECS/libsolv/libsolv.spec:13]: Thanks for the call-out; I'm not familiar with this script and how dependencies are encoded there.

I see that it already has `build_rpm_in_chroot_no_install zstd` (right near similar for `gzip`) but it's not clear to me if that's sufficient. @cwize1 would you mind taking a quick look and provide some guidance?

Comment (cwize1) [reply SPECS/libsolv/libsolv.spec:13]: That script is basically a manually specified topological ordering of the packages that make up the toolkit. Every package in the toolchain needs to have a `build_rpm_in_chroot_no_install` line. However, for packages that need to be installed to build other packages (i.e. `BuildRequires`), then you need to add a `chroot_and_install_rpms` line for that package as well. We also typically add comments describing what the build dependencies are.

For example, currently there is a line that says:

`# libsolv needs cmake`

This should probably be updated to:

`# libsolv needs cmake, libzstd`

And the following line should be added:

`chroot_and_install_rpms libzstd`

Once you think you have it right, use the `[NoArch-0-OneBranch]-Dev-PrepareBuild` pipeline to test it.

Comment (PawelWMS) [reply SPECS/libsolv/libsolv.spec:13]: And for local build debugging (if the pipeline is going to fail for some reason), you can also download the 3.0 raw toolchain, put it under `build/toolchain/toolchain_from_container.tar.gz`, then run `touch` on it to let `make` know it's fresh and run a local `make toolchain QUICK_REBUILD=y`. Alternatively, you can create this PR against `fasttrack/3.0` and that'll trigger the PR check, which builds the toolchain as the first step anyway and publishes the built RPMs.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 2]
```diff
@@ -541,8 +541,9 @@ build_rpm_in_chroot_no_install libxslt
 chroot_and_install_rpms pam
 build_rpm_in_chroot_no_install docbook-style-xsl
 
-# libsolv needs cmake
+# libsolv needs cmake, zstd-libs
 chroot_and_install_rpms cmake
+chroot_and_install_rpms zstd-libs
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:546]:
```suggestion
chroot_and_install_rpms zstd
```

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 3]
```diff
@@ -541,8 +541,9 @@ build_rpm_in_chroot_no_install libxslt
 chroot_and_install_rpms pam
 build_rpm_in_chroot_no_install docbook-style-xsl
 
-# libsolv needs cmake
+# libsolv needs cmake, zstd-libs
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:544]: In the spec you've got a BR on `libzstd-devel`, which translates to `zstd-devel`, so I think either this comment or the spec is inaccurate.

Comment (reubeno) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh]: Thanks for the feedback. I've updated the comment and arguments for consistency.

File: SPECS/libsolv/libsolv.spec (hunk 1 above)
Comment (cwize1) [reply SPECS/libsolv/libsolv.spec]: If `[AMD64-1-OneBranch]-Dev-BuildToolchain` succeeds, then you should be good.

**Review by PawelWMS — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Comment by reubeno** [comment]
@PawelWMS -- I forgot to explicitly mention, but I just wanted to confirm that the phase 1 `Dev-BuildToolchain` pipeline succeeds with the latest changes: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=648440&view=results (with `Force re-building raw toolchain` *not* selected).

**Pull request merged**